File: src/MixNetFx/Facesso.GenericControls/ucWorkGroupItemDetailsView.vb
Imports ActiveDev
Imports Facesso.Data
Imports System.Windows.Forms

Public Class ucWorkGroupItemDetailsView
    Inherits ucObjectContentDataGridView(Of WorkGroupAnalysisInfoItem)

    Protected Overrides Sub AssignValues()
        If [Object].HasData Then
            Me.Rows.Add(New Object() {"Referenzzeit", [Object].TotalReferenceIWT})
            Me.Rows.Add(New Object() {"Effektivzeit", [Object].TotalEffectiveIWT})
            Me.Rows.Add(New Object() {"Ang. Effektivzeit", [Object].TotalEffectiveIWTAdj})
            Me.Rows.Add(New Object() {"Gesamt-Pausenzeit", [Object].TotalWorkBreakTime})
            Me.Rows.Add(New Object() {"Gesamt-Ausfallzeit", [Object].TotalDownTime})
            Me.Rows.Add(New Object() {[Object].WorkGroup.IncentiveIndicatorSynonym, [Object].DegreeOfTime.ToString([Object].WorkGroup.IncentiveFormatString)})
            Me.Rows.Add(New Object() {"Ang. " & [Object].WorkGroup.IncentiveIndicatorSynonym, [Object].DegreeOfTimeAdj.ToString([Object].WorkGroup.IncentiveFormatString)})
            Me.Rows.Add(New Object() {"Ist ausgesetzt", IIf([Object].IsSuspended, "Ja", "Nein").ToString})
        Else
            Me.Rows.Add(New Object() {"Referenzzeit", "- - -"})
            Me.Rows.Add(New Object() {"Effektivzeit", "- - -"})
            Me.Rows.Add(New Object() {"Ang. Effektivzeit", "- - -"})
            Me.Rows.Add(New Object() {"Gesamt-Pausenzeit", "- - -"})
            Me.Rows.Add(New Object() {"Gesamt-Ausfallzeit", "- - -"})
            Me.Rows.Add(New Object() {[Object].WorkGroup.IncentiveIndicatorSynonym, "- - -"})
            Me.Rows.Add(New Object() {"Ang. " & [Object].WorkGroup.IncentiveIndicatorSynonym, "- - -"})
            Me.Rows.Add(New Object() {"Ist ausgesetzt", "- - -"})
        End If
    End Sub
End Class

File: src/NetFxCSharp/Facesso.GenericControls/ucWorkGroupItemDetailsView.cs
using ActiveDev;
using Facesso.Data;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace Facesso.GenericControls
{
    public class ucWorkGroupItemDetailsView : ucObjectContentDataGridView<WorkGroupAnalysisInfoItem>
    {
        protected override void AssignValues()
        {
            if (Object.HasData)
            {
                this.Rows.Add(new object[] { "Referenzzeit", Object.TotalReferenceIWT });
                this.Rows.Add(new object[] { "Effektivzeit", Object.TotalEffectiveIWT });
                this.Rows.Add(new object[] { "Ang. Effektivzeit", Object.TotalEffectiveIWTAdj });
                this.Rows.Add(new object[] { "Gesamt-Pausenzeit", Object.TotalWorkBreakTime });
                this.Rows.Add(new object[] { "Gesamt-Ausfallzeit", Object.TotalDownTime });
                this.Rows.Add(new object[] { Object.WorkGroup.IncentiveIndicatorSynonym, Object.DegreeOfTime.ToString(Object.WorkGroup.IncentiveFormatString) });
                this.Rows.Add(new object[] { "Ang. " + Object.WorkGroup.IncentiveIndicatorSynonym, Object.DegreeOfTimeAdj.ToString(Object.WorkGroup.IncentiveFormatString) });
                this.Rows.Add(new object[] { "Ist ausgesetzt", Microsoft.VisualBasic.Interaction.IIf(Object.IsSuspended, "Ja", "Nein").ToString() });
            }
            else
            {
                this.Rows.Add(new object[] { "Referenzzeit", "- - -" });
                this.Rows.Add(new object[] { "Effektivzeit", "- - -" });
                this.Rows.Add(new object[] { "Ang. Effektivzeit", "- - -" });
                this.Rows.Add(new object[] { "Gesamt-Pausenzeit", "- - -" });
                this.Rows.Add(new object[] { "Gesamt-Ausfallzeit", "- - -" });
                this.Rows.Add(new object[] { Object.WorkGroup.IncentiveIndicatorSynonym, "- - -" });
                this.Rows.Add(new object[] { "Ang. " + Object.WorkGroup.IncentiveIndicatorSynonym, "- - -" });
                this.Rows.Add(new object[] { "Ist ausgesetzt", "- - -" });
            }
        }
    }
}